### PR TITLE
Fix/aut 1906/allow score maxscore

### PR DESCRIPTION
**Pull request by shaveko**
related to https://oat-sa.atlassian.net/browse/AUT-1906

### Description
When response processing is turned off for interaction, SCORE and MAXSCORE outcome declarations are ignored and not saved into item xml, this PR allows them to be entered manually and saved

### How to test

- run unit tests
- check with related PR (https://github.com/oat-sa/tao-item-runner-qti-fe/pull/308)

### CODE REVIEW

**Review by bziondik — CHANGES_REQUESTED** [review]
From ticket, as I understand user will use property External Scored
image
let's now add to removing condition this property - if it is none, then all should works as it is
but if it Human or External Machine, then we don't remove this outcome

**Comment by shaveko** [comment]
> if it is none, then all should works as it is
> but if it Human or External Machine, then we don't remove this outcome

Thanks, I've implemented the check in `-fe` lib

File: views/js/qtiCreator/widgets/interactions/states/NoRp.js [hunk 1]
```diff
@@ -23,19 +23,7 @@ define([
...
     'use strict';
 
-    return stateFactory.create(NoRp, function initStateNoRp(){
-        var interaction = this.widget.element;
-        var item = interaction.getRootElement();
-        var rp = item.responseProcessing;
-        var rpXml = xmlRenderer.render(rp);
-
-        //remove outcome SCORE if there is no response processing
-        if(!rpXml.trim()){
```
Comment (bziondik) [inline views/js/qtiCreator/widgets/interactions/states/NoRp.js:33]: maybe we need also check here instead removing? 
```
maxScoreOutcome = item.getOutcomeDeclaration('MAXSCORE');
if(!(maxScoreOutcome.attributes && maxScoreOutcome.attributes.externalScored)) {
   item.removeOutcome('MAXSCORE');
```

**Review by bziondik — CHANGES_REQUESTED** [review]
small improvement

**Review by bziondik — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful
image
image

**Review by meefox — APPROVED** [review]
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Pull request merged**